source/host/System.cc

```diff
@@ -7,21 +7,24 @@ namespace Host {
 int KSystem::main() {
     init();
     m_sceneMgr->changeScene(0);
+    if (!m_testDirector->init()) {
+        return 1;
+    }
     run();
     return 0;
 }
 
 void KSystem::init() {
     auto *sceneCreator = new SceneCreatorDynamic;
     m_sceneMgr = new EGG::SceneManager(sceneCreator);
+
+    m_testDirector = new Test::TestDirector;
 }
 
 void KSystem::run() {
-    K_LOG("Initialized successfully!");
-    K_LOG("The program will now run in an infinite loop to test 'calc' functionality.");
-    while (true) {
+    do {
         m_sceneMgr->calc();
-    }
+    } while (m_testDirector->calc());
 }
```

Comment:
`m_testDirector->calc()` can return false under two conditions. One of them is that we're out of frames to check, and one of them is a desync. We should check `m_testDirector->sync()` to determine whether or not to return an error.